test/Microsoft.AspNetCore.Razor.Evolution.Test/RazorSyntaxTreeTest.cs

```diff
@@ -20,5 +22,39 @@ public void Parse_CanParseEmptyDocument()
             Assert.NotNull(syntaxTree);
             Assert.Empty(syntaxTree.Diagnostics);
         }
+
+        [Fact]
+        public void Parse_Persists_FilePath()
+        {
+            // Arrange
+            var filePath = "test.cshtml";
+            var source = TestRazorSourceDocument.Create("@if (true) { @if(false) { <div>@something.</div> } }", fileName: filePath);
+
+            // Act
+            var syntaxTree = RazorSyntaxTree.Parse(source);
+
+            // Assert
+            Assert.Empty(syntaxTree.Diagnostics);
+            Assert.NotNull(syntaxTree);
+
+            var spans = new List<SyntaxTreeNode>();
+            GetChildren(spans, syntaxTree.Root);
+            Assert.All(spans, node => Assert.Equal(filePath, node.Start.FilePath));
+
+            void GetChildren(List<SyntaxTreeNode> nodes, SyntaxTreeNode node)
```

Comment:
Really no good reason to do this. Plus you aren't capturing anything in the closure, so not sure if it helps being a local function.